src/NumericalAlgorithms/Spectral/Mesh.hpp

```diff
@@ -250,9 +229,25 @@ class Mesh {
   void pup(PUP::er& p);
 
  private:
-  Index<Dim> extents_{};
-  std::array<Spectral::Basis, Dim> bases_{};
-  std::array<Spectral::Quadrature, Dim> quadratures_{};
+  static Spectral::Basis extract_basis(uint8_t bits);
+  static Spectral::Quadrature extract_quadrature(uint8_t bits);
```

Comment:
[optional] Bit of a weird asymmetry in that the combining function is in an anonymous namespace but the extracting functions are private statics.  I'd make them them all anonymous.